contrib/cmd/fs-idmap/fs-idmap.go

```diff
@@ -31,20 +31,22 @@ func main() {
 	if err := cmd.Start(); err != nil {
 		log.Fatalf("failed to run the helper binary: %v", err)
 	}
+	defer func() { _ = cmd.Wait() }()
 
 	path := fmt.Sprintf("/proc/%d/ns/user", cmd.Process.Pid)
 	var userNsFile *os.File
 	if userNsFile, err = os.Open(path); err != nil {
 		log.Fatalf("unable to get user ns file descriptor: %v", err)
```

Comment:
`defer` statements don't execute on `Fatalf` because it doesn't panic nor return. This should be changed to `log.Errorf` and then `return`.

Reply:
Indeed (see https://pkg.go.dev/os#Exit)

Reply:
If you switch to using the cli library we use in all of the other contrib programs, you end up with a function that returns `error` that you can safely defer in. I would probably prefer that tbh, since it also means the binary gets a `--help` page and a bunch of other things.

Reply:
Moved it all to a different function. That way we can easily os.Exit(1) and defer calls